source/core/Variants.cpp

```diff
@@ -0,0 +1,191 @@
+/**
+Copyright (c) 2018 National Instruments Corp.
+
+This software is subject to the terms described in the LICENSE.TXT file
+
+SDG
+*/
+
+/*! \file
+*/
+
+#include "TypeDefiner.h"
+#include "ExecutionContext.h"
+#include "TypeAndDataManager.h"
+#include "TDCodecVia.h"
+#include <limits>
+#include <map>
+
+namespace Vireo
+{
+class VariantAttributeManager {
+ public:
+    typedef std::map<StringRef, StaticTypeAndDataPointer, StringRefCmp> AttributeMapType;
```

Comment:
I don't think it's a good idea to be storing the attribute map of variant data in another map.
I think the Variant should be a real type in the type manager that actually holds the type, DataPointer, and attribute map.
In addition to performance reasons of avoiding a double map lookup, I don't think this implementation works when variants are copied, since the attribute is keyed off of the original data pointer.  Copying a variant would lose its attributes.

Reply:
Created issue https://github.com/ni/VireoSDK/issues/547 to address this later.